starboard/android/shared/media_decoder.cc

```diff
@@ -219,6 +227,16 @@ void MediaDecoder::WriteInputBuffers(const InputBuffers& input_buffers) {
 void MediaDecoder::WriteEndOfStream() {
   SB_DCHECK(thread_checker_.CalledOnValidThread());
 
+  if (!SbThreadIsValid(decoder_thread_)) {
```

Comment:
Why do we need to create a decoder thread here?

Reply:
Removed it, and use |first_input_written_| in AdaptiveAudioDecoder to ensure we don't call/create codec when writing EndOfStream without any inputs.